Improvements to about.md in arithmetic-operators

SyllightTheway opened the pull request:
Added improvements to `concepts/numbers/about.md`, part of [issue #1505](https://github.com/exercism/javascript/issues/1505)

- Added explanations  about the underscore and exponential notations
- Added explanations  about the special numbers `+/-Infinity`, `NaN`, and `+/-0`

junedev reviewed (changes requested):
This is great content but it somehow ended up in the wrong file. This should go into [concepts/numbers/about.md](https://github.com/exercism/javascript/blob/main/concepts/numbers/about.md) (not into arithmetic operators).

To help you sort things into the existing content there I would recommend to call your header "Notations" "Special Notations" instead and put that header above line 22 in the existing numbers about file.
Your section "Special Numbers Values" can go below line 57.

I will take a closer look at the content once it is in the right place.

SyllightTheway commented:
Sorry for the delay, my laptop conveniently broke down and I had to wait 2 days to get it fixed. I've fixed the mistake and added them into the correct files as you requested.

junedev commented on concepts/numbers/about.md:
````diff
@@ -19,6 +19,32 @@ There is no separate integer type in common everyday use.
+The E-notation indicates a number that should be multiplied by 10 raised to a given power.
+The format of E-notation is to have a number, followed by `e` or `E`, than by the power of 10 to multiply by.
+```javascript
+num = 3.125e7; // Equals to 31250000
+// The notation essentially says, "Take 3.125 and multiply it by 10^7.
````
* `const` is missing (we always try to provide code examples that follow the recommendations how you should usually write the code, even if it would work otherwise, also we always use `const` if `let` is not required because of a re-assignment)
* You can use `// =>` to demonstrate the result of an expression, like you would see it in a REPL.
* There is a quote missing at the end.

This is how it should look like, you can then apply it for other examples where needed.

```suggestion
const num = 3.125e7;
// => 31250000
// The notation essentially says, "Take 3.125 and multiply it by 10^7".
```

junedev commented on concepts/numbers/about.md:
```diff
@@ -19,6 +19,32 @@ There is no separate integer type in common everyday use.
 
+## Special Notations
+
+There are two types of notations:
+- Exponential notation(E-notation)
+- Underscore notation
```
This section can be removed, it is fine to just have the two sections below without introduction text.

junedev commented on concepts/numbers/about.md:
````diff
@@ -19,6 +19,32 @@ There is no separate integer type in common everyday use.
+```
+
+### Underscore notation
+The underscore notaion is used to make lage numbers easier to read to the user. (The compiler will ignore the underscores)
````
```suggestion
Underscores can be used to make large numbers easier to read for the user. The compiler will completely ignore the underscores.
```

junedev commented on concepts/numbers/about.md:
````diff
@@ -19,6 +19,32 @@ There is no separate integer type in common everyday use.
+console.log(num) // => 1000000
+```
+
 Numbers may also be expressed in literal forms like `0b101`, `0o13`, `0x0A`. Learn more on numeric lexical grammar [here][lexical-grammar].
````
This line should be above the "Special Notations" header. Currently it reads like these belong to the underscore notation section.

junedev commented on concepts/numbers/about.md:
```diff
@@ -19,6 +19,32 @@ There is no separate integer type in common everyday use.
+- Exponential notation(E-notation)
+- Underscore notation
+
+### Exponential notation
```
Headers should be capitalized (besides words like `the` etc., I usually use a tool like this https://capitalizemytitle.com/ if there is a longer header).

junedev commented on concepts/numbers/about.md:
```diff
@@ -56,6 +82,68 @@ There are three types of maximum (and minimum / maximum negative) values for num
+- Two error values, `NaN` and `Infinity`.
+- Two values for zero, `+0` and `-0`.
+
+### NaN
```
```suggestion
### NaN - Not a Number
```

junedev commented on concepts/numbers/about.md:
```diff
@@ -56,6 +82,68 @@ There are three types of maximum (and minimum / maximum negative) values for num
+
+### NaN
+
+The error value `NaN`(aka "Not a Number"), is produced when:
```
```suggestion
The error value `NaN`(aka "Not a Number") is produced in the following cases.
```

junedev commented on concepts/numbers/about.md:
```diff
@@ -56,6 +82,68 @@ There are three types of maximum (and minimum / maximum negative) values for num
+
+### Infinity
+`Infinity` is an error value indicating one of two problems: 
+- A number can’t be represented because its magnitude is too large.
```
```suggestion
- A number can't be represented because its magnitude is too large.
```

junedev commented on concepts/numbers/about.md:
```diff
@@ -56,6 +82,68 @@ There are three types of maximum (and minimum / maximum negative) values for num
+ `Infinity` is larger than any other number (except `NaN`). 
+ Similarly, `-Infinity` is smaller than any other number (except `NaN`)
+
+ Additionally, the global function `sFinite()`allows you to check whether a value is an actual number (neither infinite nor NaN):
```
```suggestion
The global function `isFinite()` allows you to check whether a value is an actual number (neither infinite nor `NaN`):
```

junedev commented on concepts/numbers/about.md:
````diff
@@ -56,6 +82,68 @@ There are three types of maximum (and minimum / maximum negative) values for num
+ ```
+
+### The two zeros
+`+0` or `-0` can be produced if you represented a number, that is so small that it is indistinguishable from 0.
````
```suggestion
`+0` or `-0` are distinct numbers in JavaScript. They can be produced if you represented a number, that is so small that it is indistinguishable from 0.
```

junedev reviewed (commented):
I reviewed the content. Please don't be alarmed by the amount of comments. As mentioned before, I am overall very happy with the work you have done here. I mostly commented on minor wording or formatting issues.

Currently, the CI is failing because the markdown formatting is not quite right. You should be able to fix this by running `npx babel-node scripts/format` ([reference](https://github.com/exercism/javascript/blob/main/CONTRIBUTING.md#format)).

junedev commented:
> Sorry for the delay, my laptop conveniently broke down and I had to wait 2 days to get it fixed.

No need to apologize. Hope your laptop is alright again now. It can sometimes take a maintainer quite a while to respond as well in case we are busy with other things.

junedev commented:
There is one comment unresolved and feel free to add yourself as a contributor in the concept config: https://github.com/exercism/javascript/blob/main/concepts/numbers/.meta/config.json

junedev commented:
You should be able to fix the formatting by running `npx babel-node scripts/format`.

SleeplessByte commented on concepts/numbers/about.md:
````diff
@@ -21,6 +21,35 @@ The `bigint` type is not designed to replace the `number` type for everyday uses
+Underscores can be used to make large numbers easier to read for the user. The compiler will completely ignore the underscores.
+
+```javascript
+let num = 1_000_000; // You can read this as 1,000,000
````
```suggestion
const num = 1_000_000; // You can read this as 1,000,000
```

SleeplessByte commented on concepts/numbers/about.md:
````diff
@@ -21,6 +21,35 @@ The `bigint` type is not designed to replace the `number` type for everyday uses
+
+```javascript
+let num = 1_000_000; // You can read this as 1,000,000
+console.log(num); // => 1000000
````
```suggestion
console.log(num);
// => 1000000
```

SleeplessByte commented on concepts/numbers/about.md:
````diff
@@ -21,6 +21,35 @@ The `bigint` type is not designed to replace the `number` type for everyday uses
+
+```javascript
+const num = 1_000_000; // You can read this as 1,000,000
+console.log(num); // => 1000000
````
```suggestion
console.log(num);
// => 1000000
```

Merely for consistency with the other example earlier.

SleeplessByte commented:
Thank you!
